### ISSUE TITLE
Fix distribution get upload status api issue

### ISSUE THREAD

**Issue by wy65701436** [goharbor/harbor#17944]
Harbor introduces the chunk based repliaction since v2.7.0, but there is one issue for failure resume that is blocked by upstream distribution get status API. 

The [PR](https://github.com/distribution/distribution/commit/bad5dcb60210757eb6e9f51cee78805236fbae1c) was merged into main branch, and we should back pork it to distribution v2.8 and patch it.

**Comment by wy65701436** [comment]
move it to v2.10 since the upstream doesn't merge the PR.
